### PR TITLE
Added Delilah keyboard

### PR DESCRIPTION
**Pull request by rainkeebs**
## Description
Added support for Delilah keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/delilah/config.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
```
Comment (ridingqwerty) [inline keyboards/delilah/config.h:53]:
```suggestion
```

This is in core now, no need to put in your `config.h`

File: keyboards/delilah/rules.mk [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+CONSOLE_ENABLE = no	# Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = yes
```
Comment (ridingqwerty) [inline keyboards/delilah/rules.mk:17]:
```suggestion
BOOTMAGIC_ENABLE = lite # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes   # Mouse keys
EXTRAKEY_ENABLE = yes   # Audio control and System control
CONSOLE_ENABLE = no     # Console for debug
COMMAND_ENABLE = no     # Commands for debug and configuration
SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes       # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no   # Enable keyboard backlight functionality
AUDIO_ENABLE = no
RGBLIGHT_ENABLE = yes

```

Aligned comments, converted tabs to spaces, removed firmware size diffs as they are variable & inaccurate

File: keyboards/delilah/rules.mk [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# MCU name
+MCU = atmega32u4
+BOOTLOADER = atmel-dfu
```
Comment (ridingqwerty) [inline keyboards/delilah/rules.mk:3]:
```suggestion

#  Bootloader selection
BOOTLOADER = atmel-dfu
```

File: keyboards/delilah/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "delilah.h"
```
Comment (ridingqwerty) [inline keyboards/delilah/keymaps/via/keymap.c:16]:
```suggestion
#include QMK_KEYBOARD_H
```

Prefer the generic `QMK_KEYBOARD_H` here over the explicit header name.

File: keyboards/delilah/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+void matrix_init_user(void) {
+}
+
+void matrix_scan_user(void) {
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	return true;
+}
```
Comment (ridingqwerty) [inline keyboards/delilah/keymaps/via/keymap.c:57]:
```suggestion
```

Remove unused functions.

File: keyboards/delilah/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+void matrix_init_user(void) {
+}
+
+void matrix_scan_user(void) {
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	return true;
+}
```
Comment (ridingqwerty) [inline keyboards/delilah/keymaps/default/keymap.c:57]:
```suggestion
```

Remove unused functions.

File: keyboards/delilah/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "delilah.h"
```
Comment (ridingqwerty) [inline keyboards/delilah/keymaps/default/keymap.c:16]:
```suggestion
#include QMK_KEYBOARD_H
```

Prefer the generic `QMK_KEYBOARD_H` over the explicit header name here.

File: keyboards/delilah/delilah.h [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+	K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, \
+	K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, \
+	K200, K201, K202, K203, K204, K205, K206, K207, K208, K209,       K211, \
+	K300, K301, K302, K303,             K306,       K308, K309,       K311  \
+) { \
+	{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011 }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  KC_NO, K211 }, \
+	{ K300,  K301,  K302,  K303,  KC_NO, KC_NO, K306,  KC_NO, K308,  K309,  KC_NO, K311 }  \
+}
```
Comment (ridingqwerty) [inline keyboards/delilah/delilah.h:28]:
```suggestion
}

```

File: keyboards/delilah/readme.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+A 12.25u, traditional stagger, 40% keyboard. Designed to keep another punctuation key on base layer while still being compact.
+
+* Keyboard Maintainer: Rain
+* Hardware Supported: Delilah PCB v1.0, v1.1
+* Hardware Availability: [rainkeebs](https://www.rainkeebs.mx/)
+
+
+Make example for this keyboard (after setting up your build environment):
+
+    make delilah:default
```
Comment (ridingqwerty) [inline keyboards/delilah/readme.md:14]:
```suggestion
    qmk compile -kb delilah -kb default
```

Prefer `qmk` CLI syntax in favor of `make` syntax here.

File: keyboards/delilah/readme.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    
+To reset the keyboard, hold the top left key while plugging in, or hit the reset button near the USB port
+
+Install example for this keyboard:
+
+    make delilah:default:flash
```
Comment (ridingqwerty) [inline keyboards/delilah/readme.md:20]:
```suggestion
    qmk flash -kb delilah -km default
```

Prefer `qmk` CLI over `make` syntax here.

**Review by ridingqwerty — CHANGES_REQUESTED** [review]
Some minor suggestions.  Definitely remove the chibios files from this PR.

**Comment by rainkeebs** [comment]
> Some minor suggestions. Definitely remove the chibios files from this PR.

I'm not sure how to get the chibios files out of the PR... didn't even notice they were part of it

**Comment by rainkeebs** [comment]
opening a new PR without the changed chibios files

**Pull request closed without merging**